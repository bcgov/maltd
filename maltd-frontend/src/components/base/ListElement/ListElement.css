.large-size {
  font-size: large;
}
