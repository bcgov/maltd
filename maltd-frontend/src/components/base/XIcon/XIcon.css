.inline-block {
  display: inline-block !important;
  margin-top: -10px;
  margin-left: 20px;
}
