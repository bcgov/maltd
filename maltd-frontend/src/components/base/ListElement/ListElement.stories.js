import React from "react";
import { storiesOf } from "@storybook/react";
import ListElement from "./ListElement";

const listElement = {
  title: "MALTD"
};

storiesOf("List Element", module)
  .add("not properly added resources", () => (
    <ListElement
      listElement={{
        ...listElement,
        resources: [
          { type: "Dyn", message: "error exists", status: "not-member" },
          { type: "Share", message: null, status: "member" }
        ]
      }}
    />
  ))
  .add("user", () => (
    <ListElement
      listElement={{
        ...listElement,
        title: "User name",
        description: "user@gov.bc.ca"
      }}
    />
  ))
  .add("properly added resources", () => (
    <ListElement
      listElement={{
        ...listElement,
        resources: [
          { type: "Dyn", status: "member" },
          { type: "Share", status: "member" }
        ]
      }}
    />
  ))
  .add("empty", () => (
    <ListElement listElement={{ ...listElement, title: undefined }} />
  ));
