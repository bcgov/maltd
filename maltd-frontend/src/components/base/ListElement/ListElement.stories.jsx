import React from "react";
import { storiesOf } from "@storybook/react";
import ListElement from "./ListElement.jsx";

const listElement = {
  title: "MALTD"
};

storiesOf("List Element", module)
  .add("current project", () => <ListElement listElement={listElement} />)
  .add("user", () => (
    <ListElement
      listElement={{
        ...listElement,
        title: "User name",
        description: "user@gov.bc.ca"
      }}
    />
  ))
  .add("empty", () => (
    <ListElement listElement={{ ...listElement, title: undefined }} />
  ));
