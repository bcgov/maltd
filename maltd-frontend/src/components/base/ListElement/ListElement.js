import React from "react";
import PropTypes from "prop-types";
import XIcon from "../XIcon/XIcon";
import "./ListElement.css";

export default function ListElement({
  listElement: { title, description, resources, id },
  onXClick
}) {
  let memberOfResources = "";
  let notMemberOfResources = "";
  let message;

  if (resources && resources.length > 0) {
    resources.forEach((resource, index) => {
      if (resource.message) {
        notMemberOfResources += `${resource.type}`;
        message = resource.message;
      } else if (index !== resources.length - 1) {
        memberOfResources += `${resource.type}, `;
      } else {
        memberOfResources += resource.type;
      }
    });
  }

  if (title && (memberOfResources || description)) {
    return (
      <React.Fragment>
        <div className="project-div">
          <div className="project-list-item" id="project-info">
            <strong className="large-size">{title}</strong>
            {description && (
              <p className="project-list-item" id="email-info">
                {description}
              </p>
            )}
            {memberOfResources && (
              <p className="project-list-item" id="member-resources">
                Member: {memberOfResources}
              </p>
            )}
            {notMemberOfResources && (
              <React.Fragment>
                <p className="project-list-item" id="member-resources">
                  Not member: {notMemberOfResources}
                </p>
                <p className="project-list-item" id="member-resources">
                  <b>Error:</b> {message}
                </p>
              </React.Fragment>
            )}
          </div>
          <div>
            {typeof onXClick === "function" && (
              <XIcon id={id} onClick={onXClick} />
            )}
          </div>
        </div>
      </React.Fragment>
    );
  }

  return (
    <div>
      <p>No projects</p>
    </div>
  );
}

ListElement.propTypes = {
  listElement: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    resources: PropTypes.array,
    id: PropTypes.string
  }).isRequired,
  onXClick: PropTypes.func
};

ListElement.defaultProps = {
  onXClick: null
};
