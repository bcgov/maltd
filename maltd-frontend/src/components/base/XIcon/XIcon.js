import React from "react";
import PropTypes from "prop-types";
import Icon from "../../../img/x-icon.PNG";
import "./XIcon.css";

export default function XIcon({ onClick, id }) {
  return (
    <div
      id="main"
      role="button"
      tabIndex="0"
      onClick={() => onClick(id)}
      onKeyDown={() => onClick(id)}
      className="inline-block"
    >
      <img src={Icon} alt="x icon" width="25px" height="25px" />
    </div>
  );
}

XIcon.propTypes = {
  onClick: PropTypes.func.isRequired,
  id: PropTypes.string
};
