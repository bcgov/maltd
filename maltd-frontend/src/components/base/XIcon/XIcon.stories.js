import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import "bootstrap/dist/css/bootstrap.css";
import XIcon from "./XIcon";

storiesOf("XIcon", module).add("default", () => (
  <XIcon id="12" onClick={action("x icon clicked")} />
));
