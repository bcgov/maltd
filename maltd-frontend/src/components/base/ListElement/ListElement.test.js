import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import ListElement from "./ListElement";
import XIcon from "../XIcon/XIcon";

Enzyme.configure({ adapter: new Adapter() });

describe("List Element", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(
      <ListElement
        listElement={{
          title: "title",
          description: "description"
        }}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Component displays XIcon when it should", () => {
    const testState = { isClicked: false };
    const onClick = () => {
      testState.isClicked = true;
    };

    const wrapper = shallow(
      <ListElement
        listElement={{
          title: "title",
          description: "description",
          id: "12"
        }}
        onXClick={onClick}
      />
    );

    expect(wrapper.contains(<XIcon id="12" onClick={onClick} />)).toBe(true);
  });

  test("Component does not display XIcon when it should not", () => {
    const testState = { isClicked: false };
    const onClick = () => {
      testState.isClicked = true;
    };

    const wrapper = shallow(
      <ListElement
        listElement={{
          title: "title",
          description: "description",
          id: "12"
        }}
        onXClick={null}
      />
    );

    expect(wrapper.contains(<XIcon id="12" onClick={onClick} />)).toBe(false);
  });

  test("Component generates the member of resources string appropriately", () => {
    const wrapper = shallow(
      <ListElement
        listElement={{
          title: "title",
          description: "description",
          resources: [{ type: "Dyn" }, { type: "Share" }]
        }}
      />
    );

    const rows = wrapper.find("#member-resources");

    expect(rows.length).toEqual(1);
    expect(rows.text()).toEqual("Member: Dyn, Share");
  });
});
