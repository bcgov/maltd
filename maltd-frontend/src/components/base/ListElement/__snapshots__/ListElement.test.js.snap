// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Element Component renders as expected 1`] = `
<div
  className="project-div"
>
  <div
    className="project-list-item"
    id="project-info"
  >
    <strong
      className="large-size"
    >
      title
    </strong>
    <p
      className="project-list-item"
      id="email-info"
    >
      description
    </p>
    
    
  </div>
  <div />
</div>
`;
